src/random.cpp

```diff
@@ -143,6 +143,34 @@ static bool GetHardwareRand(unsigned char* ent32) noexcept {
     return false;
 }
 
+/** Use repeated SHA512 to strengthen the randomness in seed32, and feed into hasher. */
+static void Strengthen(const unsigned char* seed32, int microseconds, CSHA512& hasher) noexcept
+{
+    CSHA512 inner_hasher;
+    inner_hasher.Write(seed32, 32);
```

Comment:
Unless I'm misunderstanding it, the data collected by SeedStartup doesn't get strengthened in the first run of the strenghtener. This seems pretty suboptimal since SeedStartup is the obvious place to go get most of the weak pseudo-entropy (like filesystem data, network interface data, etc.) which is the stuff that most benefits from strengthening.

Reply:
Good point, fixed.

It now extracts using the hasher state combined with the RNG.

So overall what happens in SeedStartup is:
* Gather entropy using SeedSlow/RandAddSeedPerfmon/SeedStrengthen, in a hasher `E`. `E = SeedStuff()`
* Mix E together with the (in this case empty RNG state `R0`), and split the result in a seed and the new RNG state, so `[S,R1] = SHA512(E || R0)`
* `S` gets strengthened, `S' = SHA512(SHA512(...(S)))`
* `S'` is added to `E`, and mixed into the RNG: `[_,R2] = SHA512(E || S' || R1)`.

It's a bit weird in that it uses `E` both in construction `R1` and `R2`, but I don't see how that can hurt.